### PR TITLE
Handle relative paths and `~` character in `--target-path` of `new` command

### PR DESCRIPTION
**Pull request by joshuayoes**
## Please verify the following:

- [x] `yarn ci:test` **jest** tests pass with new tests, if relevant
- [x] `README.md` has been updated with your changes, if relevant

## Describe your PR
Closes https://github.com/infinitered/ignite/issues/2141

### CODE REVIEW

File: src/commands/new.ts [hunk 1]
```diff
@@ -195,8 +195,12 @@ export default {
     // #endregion
 
     // #region Project Path
+    const parsePath = (p: string) =>
+      p.startsWith("~") ? path(p.replace("~", filesystem.homedir())) : path(p)
```
Comment (joshuayoes) [inline src/commands/new.ts:199]: For some reason, `filesystem.path` does not seem to like `~/Code/PizzaApp`. So instead, we are replacing the `~` with the actual home dir path

File: src/commands/new.ts [hunk 2]
```diff
@@ -195,8 +195,12 @@ export default {
...
     const defaultTargetPath = path(projectName)
-    let targetPath = useDefault(options.targetPath) ? defaultTargetPath : options.targetPath
+    let targetPath = useDefault(options.targetPath)
+      ? defaultTargetPath
+      : parsePath(options.targetPath)
```
Comment (joshuayoes) [inline src/commands/new.ts:203]: Wrapping flag args in `filesystem.path` expands `./Code/PizzaApp` to `~/Code/PizzaApp`

File: src/commands/new.ts [hunk 3]
```diff
@@ -397,7 +401,7 @@ export default {
     // Release Ignite installs have the boilerplate's .gitignore in .gitignore.template
     // (see https://github.com/npm/npm/issues/3763); development Ignite still
     // has it in .gitignore. Copy it from one or the other.
-    const targetIgnorePath = log(path(process.cwd(), ".gitignore"))
+    const targetIgnorePath = log(path(boilerplatePath, ".gitignore"))
```
Comment (joshuayoes) [inline src/commands/new.ts:404]: This is a regression that wasn't caught earlier

File: src/commands/new.ts [hunk 4]
```diff
@@ -501,7 +505,7 @@ export default {
     // #endregion
 
     // #region Cache dependencies
-    if (shouldFreshInstallDeps && cacheExists === false) {
+    if (shouldFreshInstallDeps && cacheExists === false && useCache) {
```
Comment (joshuayoes) [inline src/commands/new.ts:508]: Even though `useCache` is `false` by default now, this was still getting copied. Added `useCache` check to only copy if `--use-cache` is `true`

**Review by jamonholmgren — APPROVED** [review]
looks great

**Pull request merged**